### DISCUSSION TITLE
Property 'toBeEmptyDOMElement' does not exist on type 'Assertion<HTMLElement>'

### DISCUSSION THREAD

**Discussion by agriffis** [vitest-dev/vitest#6224] [Q&A]
Hi! I'm stymied by this:

![image](https://github.com/user-attachments/assets/a2dcb115-22ca-4b90-8ee8-9b957a8f5cba)

I've tried both global mode and explicit mode.

Other assertions in the expect suite seem fine.

I'm just not sure why TypeScript doesn't like this particular assertion.

Okay, here's a guess: I'm also using Storybook and it's somehow supplying additional types that are conflicting. But I'm not sure how that works.

Any suggestions?

**Comment by sheremet-va** [comment]
Is this browser mode? Or are you using `@testing-library/jest-dom`?

**Reply by agriffis** [reply]
Hello! I'm using `@testing-library/jest-dom` version `6.4.8`

**Reply by agriffis** [reply]
Mixed progress...

I have in my `vitest.setup.ts`: `import '@testing-library/jest-dom/vitest'` which came from documentation I can no longer find. So I've tried a few different ideas:

|       | no import                     | `import '@testing-library/jest-dom/vitest'` | `import '@testing-library/jest-dom'`                                         |
| ----- | ----------------------------- | ------------------------------------------- | ---------------------------------------------------------------------------- |
| types | fail on `toBeEmptyDOMElement` | fail on `toBeEmptyDOMElement`               | types work!                                                                  |
| tests | fail on `toBeEmptyDOMElement` | tests work!                                 | all tests fail, problems with jest-dom internals looking for global `expect` |

I hate the feeling that I'm stabbing at this randomly :frowning:
